### ISSUE TITLE
Error when missing.values=""

### ISSUE THREAD

**Issue by bixiou** [melff/memisc#62]
I have recently changed computer.
I have installed R (4.2.1), RStudio (2022.07.1 Build 554) and memisc 0.99.22.
When I run the following code:
```
test <- as.item("a", missing.values="")
test[test!=""]
```
I get this error: `Error: C stack usage  15923312 is too close to the limit`.
It's weird, because on my previous computer, this type of code didn't raise any issue.
Can you help me debug?

**Comment by bixiou** [comment]
PS: a cumbersome workaround is to replace the the second line by `include.missings(test)[test!=""]`.

**Comment by bixiou** [comment]
Any fix? Now I get the same error message where I use `is.missing`. It's very annoying.

**Comment by melff** [comment]
This problem does not occur with the current version (0.99.30) of memisc. Hence this is to be considered a non-issue.

**Comment by bixiou** [comment]
Well, I am using the version 0.99.22 because it has some useful features that have later disappeared. 
Could you help me identify the problem so that I can try to code a patch?
I've seen that the last release fixed a bug with an infinite recursion: this seems to be the same (type of) bug.

**Comment by melff** [comment]
I extracted the relevant commit as a diff file. I hope that helps.

[patch-infinite-recursion.txt](https://github.com/melff/memisc/files/9675247/patch-infinite-recursion.txt)

BTW: Which useful features have disappeared? If those features I are useful to others, I shall reimplement them.

**Comment by bixiou** [comment]
Thank you!

Apparently it was an unintended feature, see [this issue](https://github.com/melff/memisc/issues/52).
By turning strings defined as missing values (e.g. `PNR`) to strings when applying `as.character`, : it allows me to run regressions without dropping data ('PNR' are considered as a response category, which it is) while allowing me to treat these 'PNR' as missing in the analysis (i.e. have a special status for descriptive stats, graphs...). (With the newer versions, these missing values are turned to `NA`.)

**Comment by bixiou** [comment]
Arf, applying the patch did not work, I still get the error :-/

I enter an infinite recursion. Here is the log with `traceback()` after the running code above:

```
18: [... it gets in an infinite loop beyond that]
17: is.missing(x)
16: as.vector(x, mode)
15: as.vector(x, mode)
14: as.vector(x)
13: mtfrm.default(new("character.item", .Data = "a", value.labels = NULL, 
        value.filter = new("missing.values", range = NULL, filter = ""), 
        measurement = "nominal", annotation = new("annotation", .Data = character(0))))
12: mtfrm(new("character.item", .Data = "a", value.labels = NULL, 
        value.filter = new("missing.values", range = NULL, filter = ""), 
        measurement = "nominal", annotation = new("annotation", .Data = character(0))))
11: x %in% filter@filter
10: x %in% filter@filter
9: is.missing2(x, filter)
8: is.missing(x)
7: is.missing(x)
6: .local(x, ...)
5: format(x, use.labels = use.labels, ...)
4: format(x, use.labels = use.labels, ...)
3: print.item.vector(object, width = getOption("width"), compress = TRUE)
2: (new("standardGeneric", .Data = function (object) 
   standardGeneric("show"), generic = structure("show", package = "methods"), 
       package = "methods", group = list(), valueClass = character(0), 
       signature = structure("object", simpleOnly = TRUE), default = new("derivedDefaultMethod", 
           .Data = function (object) 
           showDefault(object), target = new("signature", .Data = "ANY", 
               names = "object", package = "methods"), defined = new("signature", 
               .Data = "ANY", names = "object", package = "methods"), 
           generic = structure("show", package = "methods")), skeleton = (new("derivedDefaultMethod", 
           .Data = function (object) 
           showDefault(object), target = new("signature", .Data = "ANY", 
               names = "object", package = "methods"), defined = new("signature", 
               .Data = "ANY", names = "object", package = "methods"), 
           generic = structure("show", package = "methods")))(object)))(new("character.item", 
       .Data = "a", value.labels = NULL, value.filter = new("missing.values", 
           range = NULL, filter = ""), measurement = "nominal", 
       annotation = new("annotation", .Data = character(0))))
1: (new("standardGeneric", .Data = function (object) 
   standardGeneric("show"), generic = structure("show", package = "methods"), 
       package = "methods", group = list(), valueClass = character(0), 
       signature = structure("object", simpleOnly = TRUE), default = new("derivedDefaultMethod", 
           .Data = function (object) 
           showDefault(object), target = new("signature", .Data = "ANY", 
               names = "object", package = "methods"), defined = new("signature", 
               .Data = "ANY", names = "object", package = "methods"), 
           generic = structure("show", package = "methods")), skeleton = (new("derivedDefaultMethod", 
           .Data = function (object) 
           showDefault(object), target = new("signature", .Data = "ANY", 
               names = "object", package = "methods"), defined = new("signature", 
               .Data = "ANY", names = "object", package = "methods"), 
           generic = structure("show", package = "methods")))(object)))(new("character.item", 
       .Data = "a", value.labels = NULL, value.filter = new("missing.values", 
           range = NULL, filter = ""), measurement = "nominal", 
       annotation = new("annotation", .Data = character(0))))
```

**Comment by bixiou** [comment]
It worked! Thanks a lot!!!
In case it can be useful to others, here is the patched version 0.99.22: [memisc_0.99.22.tar.gz](https://github.com/melff/memisc/files/9690453/memisc_0.99.22.tar.gz).
